### PR TITLE
Add PayoutTxValidation

### PR DESCRIPTION
**Pull request by HenrikJannsen**
## Release Notes

* **Bug Fixes**
  * Enhanced payout transaction validation with improved error handling and explicit logging of validation failures instead of silent exceptions.

* **Tests**
  * Added comprehensive test suite covering payout transaction validation scenarios, including edge cases and failure conditions.

### CODE REVIEW

File: core/src/main/java/bisq/core/trade/validation/PayoutTxValidation.java [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+                                                 TransactionOutput depositOutput,
+                                                 Script redeemScript,
+                                                 byte[] buyerMultiSigPubKey,
+                                                 byte[] sellerMultiSigPubKey) {
+        Script depositOutputScript = depositOutput.getScriptPubKey();
+        boolean isSegwitPayout = Arrays.equals(depositOutputScript.getProgram(),
+                ScriptBuilder.createP2WSHOutputScript(redeemScript).getProgram());
+        checkArgument(isSegwitPayout,
+                "depositTx output 0 must be the expected P2WSH 2-of-2 multisig script");
+
```
[the remark this answers is not in the document]
Comment (HenrikJannsen) [reply core/src/main/java/bisq/core/trade/validation/PayoutTxValidation.java:203]: No we do not support support legacy P2SH. TradeWalletService might have old methods used from old versions. Should be cleaned up.

File: core/src/main/java/bisq/core/trade/validation/PayoutTxValidation.java [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+                    redeemScript,
+                    depositOutput.getValue(),
+                    Transaction.SigHash.ALL,
+                    false);
+            ECKey buyerKey = ECKey.fromPublicOnly(buyerMultiSigPubKey);
+            ECKey sellerKey = ECKey.fromPublicOnly(sellerMultiSigPubKey);
+            checkArgument(buyerKey.verify(sigHash, buyerSignature) && sellerKey.verify(sigHash, sellerSignature),
+                    "payoutTx witness signatures are invalid");
+        } catch (Throwable t) {
+            throw new IllegalArgumentException("payoutTx witness signatures are invalid", t);
```
[the remark this answers is not in the document]
Comment (HenrikJannsen) [reply core/src/main/java/bisq/core/trade/validation/PayoutTxValidation.java:255]: Will address in new PR

File: core/src/test/java/bisq/core/trade/validation/PayoutTxValidationTest.java [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+                false);
+        TransactionSignature buyerSignature = new TransactionSignature(BUYER_MULTI_SIG_KEY.sign(sigHash),
+                Transaction.SigHash.ALL,
+                false);
+        TransactionSignature sellerSignature = new TransactionSignature(SELLER_MULTI_SIG_KEY.sign(sigHash),
+                Transaction.SigHash.ALL,
+                false);
+        input.setScriptSig(ScriptBuilder.createEmpty());
+        input.setWitness(TransactionWitness.redeemP2WSH(REDEEM_SCRIPT, sellerSignature, buyerSignature));
+        return payoutTx;
```
[the remark this answers is not in the document]
Comment (HenrikJannsen) [reply core/src/test/java/bisq/core/trade/validation/PayoutTxValidationTest.java:330]: Will address in next PR
